filebeat/module/auditd/log/ingest/pipeline.yml

```diff
@@ -2102,6 +2102,29 @@ processors:
     ignore_failure: true
     field: auditd.log.msg
     target_field: message
+- script:
+    lang: painless
+    description: Extracts process information from execve calls
+    if: 'ctx.process?.args_count != null && ctx.auditd?.log != null'
+    source: >-
+      long argc = ctx.process.args_count;
+      List args = new ArrayList();
+      def[] fmt = new def[] {0};
+      for (long i=0; i<argc; i++) {
+        fmt[0] = i;
+        String fld = String.format("a%d", fmt);
+        def arg = ctx.auditd.log.remove(fld);
+        if (arg == null) break;
+        args.add(arg);
```

Comment:
Can we retain the original argument order? At the moment we are retaining the args in lexical order.

Reply:
This is caused by the test_modules.py sorting all array fields before comparison, as some array fields can be generated in a different order each run. 

https://github.com/elastic/beats/blob/405c3428b9f5c135c78058aa2c183feb3200edd9/filebeat/tests/system/test_modules.py#L200-L202

I've added some configuration to the testing so that we can skip the sorting step for some fields, for now just `process.args`.